halo2_proofs/src/plonk/circuit.rs

```diff
@@ -334,6 +335,68 @@ impl TableColumn {
     }
 }
 
+/// Users of halo2 cannot use this struct.
```

Comment:
Not sure what this means exactly. The struct is `pub`. So IIUC you say it because the `usize` inside of the tuplestruct should never be accessed by the user?

Reply:
Yes, this was @moreSocratic's comment, but my interpretation is:

Since the `usize` is private, `DynamicTable` must always be constructed by calling `create_dynamic_table()`, and cannot be directly instantiated with an arbitrary index.